main.go

```diff
@@ -1,68 +1,160 @@
 package main
 
 import (
-	"os"
-	"fmt"
-	"net"
-	"strings"
+	"bufio"
+	"flag"
+	"io"
+	"log"
 	"math/rand"
-	"io/ioutil"
+	"net"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
 )
 
-func randomQuote() ([]byte, error) {
-	bs, err := ioutil.ReadFile(os.Args[2])
+type quoteServer struct {
+	quotes     []string
+	randPool   sync.Pool // sync.Pool<*rand.Rand>
```

Comment:
rand.Rand is not safe for concurrent use, that leaves us two options, either have one rand.Rand per goroutine, or have one shared between all goroutine and then make sure only one goroutine can access the rand.Rand at a time.

I  chose to have a rand.Rand per goroutine (or thread) which is approximately what using sync.Pool achieves.

Reply:
Is there a way I can know which functions and methods are safe to use concurrently? Theres nothing written in the [documentation](https://golang.org/pkg/math/rand/#Intn).

Reply:
The global rand functions use a mutex to protect concurrent access to the shared source.

https://golang.org/pkg/math/rand/#NewSource

Genrally, functions are thread safe, wheres methods are not, unless documented otherwise.